packages/gatsby/src/utils/validate-plugin-options.js

```diff
@@ -0,0 +1,23 @@
+exports.validatePluginOptions = (res, options = {}) => {
+  let chain = Promise.resolve(res)
+  if (res && res.validate) {
+    chain = res.validate(options, {
```

Comment:
There's a trade off here -- I think this is a best practice (e.g. we'll want to capture _all_ errors and allow unknown) but some plugins would possibly want more control over this.